lib/chef/resource/aws_vpc.rb

```diff
@@ -134,7 +133,9 @@ class Chef::Resource::AwsVpc < Chef::Provisioning::AWSDriver::AWSResourceWithEnt
 
   def aws_object
     driver, id = get_driver_and_id
-    result = driver.ec2.vpcs[id] if id
-    result && result.exists? ? result : nil
+    ec2_resource = ::Aws::EC2::Resource.new(driver.ec2)
+    result = ec2_resource.vpc(id) if id
+    result && result.data ? result : nil
```

Comment:
The exception handling should be done for this block which was removed in the previous commit.